packages/libs/src/sdk/services/Auth/UserAuth.ts

```diff
@@ -0,0 +1,149 @@
+import { GetFn, Hedgehog, SetAuthFn, SetUserFn } from '@audius/hedgehog'
+import { keccak_256 } from '@noble/hashes/sha3'
+import * as secp from '@noble/secp256k1'
+import fetch from 'cross-fetch'
+import { EIP712TypedData, MessageData, signTypedData } from 'eth-sig-util'
+
+import { mergeConfigWithDefaults } from '../../utils/mergeConfigs'
+
+import { defaultUserAuthConfig } from './constants'
+import type { UserAuthConfig, AuthService } from './types'
+
+export class UserAuth implements AuthService {
+  /**
+   * Configuration passed in by the consumer (with defaults)
+   */
+  private config: UserAuthConfig
+
+  /**
+   * Hedgehog instance to handle auth
+   */
+  private readonly hedgehog: Hedgehog
+
+  constructor(config?: UserAuthConfig) {
+    this.config = mergeConfigWithDefaults(config, defaultUserAuthConfig)
+
+    const get: GetFn = async ({ lookupKey, email, otp }) => {
+      const params = new URLSearchParams({
+        lookupKey,
+        username: email as string
+      })
+      if (otp) {
+        params.append('otp', otp as string)
+      }
+      const response = await fetch(
+        `${this.config.identityService}/authentication?${params}`
+      )
+      return response.json()
+    }
+
+    const setAuth: SetAuthFn = async (params) => {
+      return fetch(`${this.config.identityService}/authentication`, {
+        method: 'post',
+        body: JSON.stringify(params)
+      })
+    }
+
+    const setUser: SetUserFn = async (params) => {
+      return fetch(`${this.config.identityService}/user`, {
+        method: 'post',
+        body: JSON.stringify(params)
+      })
+    }
+
+    this.hedgehog = new Hedgehog(
+      get,
+      setAuth,
+      setUser,
+      config?.useLocalStorage ?? true
+    )
+    this.config.localStorage.then((ls) => (this.hedgehog.localStorage = ls))
+  }
+
+  signUp = async ({ email, password }: { email: string; password: string }) => {
+    await this.hedgehog.waitUntilReady()
+    const wallet = await this.hedgehog.signUp({ username: email, password })
+    return wallet
+  }
+
+  signIn = async ({
+    email,
+    password,
+    otp
+  }: {
+    email: string
+    password: string
+    otp?: string
+  }) => {
+    await this.hedgehog.waitUntilReady()
+    const wallet = await this.hedgehog.login({
```

Comment:
maybe a "nice to have" here would be a special error type for OTP requests, but it's not important rn